doc/adr/2022-11-16-option-precedence-hierarchy.md

```diff
@@ -0,0 +1,28 @@
+**Title**: Establish how multiple values for the same option are reconciled.
+
+**Status**: proposed
+
+**Context**: There are a variety of ways that options can be customized for a
+given rpc call. Both `Client` and `Connection` classes can be created with a set
+of `google::cloud::Options`. Individual function calls in a `Client` can be
+passed `google::cloud::Options`. Some environment variables can be set that
+such that the variable value sets an Option associated with that environment
+variable. Lastly, GCS `Request` types inherit from various Option templates
```

Comment:
maybe "Lastly, the storage library precedes the `google::cloud::Options` mechanism. Many of the functions in this library consume a parameter pack, the parameter pack sets optional values for the RPCs used in the function."  How these are implemented in the `*Request` classes is neither here not there.